pubtools/pulplib/_impl/client/client.py

```diff
@@ -179,6 +182,28 @@ def search_repository(self, criteria=None):
             response_f, lambda data: self._handle_page(Repository, search, data)
         )
 
+    def _do_upload_file(self, upload_id, file_obj):
+        def do_next_upload(checksum, size):
+            data = file_obj.read(self._CHUNK_SIZE)
+            if data:
+                checksum.update(data)
+                return f_flat_map(
+                    self._do_upload(data, upload_id, size),
+                    lambda _: do_next_upload(checksum, size + len(data)),
+                )
+            else:
+                # nothing more to upload, return checksum and size
+                if is_string:
+                    file_obj.close()
+                return f_return((checksum.hexdigest(), size))
+
+        is_string = isinstance(file_obj, six.string_types)
+        if is_string:
+            file_obj = open(file_obj, "rb")
+
+        LOG.info("Uploading %s to Pulp", file_obj)
```

Comment:
This doesn't seem appropriate for an INFO log due to the fact that file_obj can be some object with no meaning to end-users.

For example I think this would end up with messages like these in Pub logs:

```
[INFO] Uploading <_io.BufferedReader name='/some/file'> to Pulp
```
Would it make more sense to log the name?  Or make this DEBUG if you wanted it for debugging purposes.